pkg/apiserver/interfaces/api/repository.go

```diff
@@ -69,19 +69,41 @@ func (h repository) GetWebServiceRoute() *restful.WebService {
 		Writes([]string{}))
 
 	// List available chart versions
-	ws.Route(ws.GET("/charts/{chart}/versions").To(h.listVersions).
+	ws.Route(ws.GET("/chart/versions").To(h.listVersions).
 		Doc("list versions").
 		Metadata(restfulspec.KeyOpenAPITags, tags).
+		Param(ws.QueryParameter("chart", "helm chart").DataType("string").Required(true)).
+		Param(ws.QueryParameter("repoUrl", "helm repository url").DataType("string").Required(true)).
+		Param(ws.QueryParameter("secretName", "secret of the repo").DataType("string")).
+		Returns(200, "OK", v1.ChartVersionListResponse{}).
+		Returns(400, "Bad Request", bcode.Bcode{}).
+		Writes([]string{}))
+
+	ws.Route(ws.GET("/charts/{chart}/versions").To(h.listVersions).
+		Doc("list versions").Deprecate().
+		Metadata(restfulspec.KeyOpenAPITags, tags).
 		Param(ws.QueryParameter("repoUrl", "helm repository url").DataType("string")).
 		Param(ws.QueryParameter("secretName", "secret of the repo").DataType("string")).
 		Returns(200, "OK", v1.ChartVersionListResponse{}).
 		Returns(400, "Bad Request", bcode.Bcode{}).
 		Writes([]string{}))
 
 	// List available chart versions
-	ws.Route(ws.GET("/charts/{chart}/versions/{version}/values").To(h.chartValues).
+	ws.Route(ws.GET("/chart/values").To(h.chartValues).
 		Doc("get chart value").
 		Metadata(restfulspec.KeyOpenAPITags, tags).
+		Param(ws.QueryParameter("chart", "helm chart").DataType("string").Required(true)).
+		Param(ws.QueryParameter("version", "helm chart version").DataType("string").Required(true)).
+		Param(ws.QueryParameter("repoUrl", "helm repository url").DataType("string").Required(true)).
+		Param(ws.QueryParameter("repoType", "helm repository type").DataType("string").Required(true)).
+		Param(ws.QueryParameter("secretName", "secret of the repo").DataType("string")).
+		Returns(200, "OK", "").
```

Comment:
Return the string is not a good idea. The best result must be a JSON struct, so, keep the parsed values and same time return the YAML string. For example:

```
{
  "values": {},
  "content": "the content of the value.yaml file"
}
```

Reply:
Wrap the string in a json struct.